critiquebrainz/db/review.py

```diff
@@ -0,0 +1,608 @@
+    """
+    filters = []
+    filter_data = {}
+    inc_drafts = kwargs.pop('inc_drafts', None)
+    if not inc_drafts:
+        filters.append("is_draft = :is_draft")
+        filter_data["is_draft"] = False
+
+    inc_hidden = kwargs.pop('inc_hidden', None)
+    if not inc_hidden:
+        filters.append("is_hidden = :is_hidden")
+        filter_data["is_hidden"] = False
+
+    # FILTERING
+    entity_id = kwargs.pop('entity_id', None)
+    if entity_id is not None:
+        filters.append("entity_id = :entity_id")
+        filter_data["entity_id"] = entity_id
+
+    entity_type = kwargs.pop('entity_type', None)
+    if entity_type is not None:
+        filters.append("entity_type = :entity_type")
+        filter_data["entity_type"] = entity_type
+
+    license_id = kwargs.pop('license_id', None)
+    if license_id is not None:
+        filters.append("license_id = :license_id")
+        filter_data["license_id"] = license_id
+
+    user_id = kwargs.pop('user_id', None)
+    if user_id is not None:
+        filters.append("review.user_id = :user_id")
+        filter_data["user_id"] = user_id
+
+    exclude = kwargs.pop('exclude', None)
+    if exclude is not None:
+        filters.append("review.id NOT IN :exclude")
+        filter_data["exclude"] = tuple(exclude)
+
+    language = kwargs.pop('language', None)
+    if language is not None:
+        filters.append("language = :language")
+        filter_data["language"] = language
+
+    filterstr = " AND ".join(filters)
+    if filterstr:
+        filterstr = " WHERE " + filterstr
+
+    query = sqlalchemy.text("""
+        SELECT COUNT(*)
+          FROM review
+            {}
+        """.format(filterstr))
+
+    with db.engine.connect() as connection:
+        result = connection.execute(query, filter_data)
+        count = result.fetchone()[0]
+    sort = kwargs.pop("sort", None)
+    order_by_clause = str()
+
+    if sort == 'rating':
+        order_by_clause = """
+            ORDER BY rating DESC
+        """
+    elif sort == 'created':
+        order_by_clause = """
+            ORDER BY created DESC
+        """
+    elif sort == 'random':
+        order_by_clause = """
+            ORDER BY RANDOM()
+        """
+    # Note that all revisions' votes are considered in these ratings
+    query = sqlalchemy.text("""
+        SELECT review.id,
+               entity_id,
+               entity_type,
+               edits,
+               is_draft,
+               is_hidden,
+               license_id,
+               language,
+               review.user_id,
+               display_name,
+               show_gravatar,
+               is_blocked,
+               email,
+               "user".created as user_created,
+               musicbrainz_id,
+               MIN(revision.timestamp) as created,
+               SUM(
+                CASE
+                 WHEN vote='t' THEN 1
+                 ELSE 0
+                END
+               ) AS votes_positive_count,
```

Comment:
A couple of problems here. The goal of splitting parts of a query into multiple rows is to make it more readable. It doesn't mean that everything needs to be like that. This query doesn't even fit on my screen.

Statement like this is still readable:
```sql
SUM(
    CASE WHEN vote='t' THEN 1 ELSE 0 END
) AS votes_positive_count,
```